server.go

```diff
@@ -1015,8 +1015,38 @@ func (s *Server) applyDropDatabase(m *messaging.Message) (err error) {
 	// Remove from metastore.
 	err = s.meta.mustUpdate(m.Index, func(tx *metatx) error { return tx.dropDatabase(c.Name) })
 
+	db := s.databases[c.Name]
+	for _, rp := range db.policies {
+		for _, sg := range rp.shardGroups {
+			for _, sh := range sg.Shards {
+
+				// if we have this shard locally, close and remove it
+				if sh.store != nil {
+					// close topic readers/heartbeaters/etc. connections
+					err := s.client.CloseConn(sh.ID)
+					if err != nil {
+						panic(err)
```

Comment:
Cool. Panic works for me.

On Tuesday, April 21, 2015, Jason Wilder <notifications@github.com> wrote:

> In server.go
> <https://github.com/influxdb/influxdb/pull/2353#discussion_r28811098>:
>
> > @@ -1015,8 +1015,38 @@ func (s *Server) applyDropDatabase(m *messaging.Message) (err error) {
> >  	// Remove from metastore.
> >  	err = s.meta.mustUpdate(m.Index, func(tx *metatx) error { return tx.dropDatabase(c.Name) })
> >
> > +	db := s.databases[c.Name]
> > +	for _, rp := range db.policies {
> > +		for _, sg := range rp.shardGroups {
> > +			for _, sh := range sg.Shards {
> > +
> > +				// if we have this shard locally, close and remove it
> > +				if sh.store != nil {
> > +					// close topic readers/heartbeaters/etc. connections
> > +					err := s.client.CloseConn(sh.ID)
> > +					if err != nil {
> > +						panic(err)
>
> These shouldn't ever happen so I used panic so we can get get a line
> number and more context if they did. Not sure how best to handle this
> though because we've dropped the database from the metastore but we're
> failing to cleanup resources.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/2353/files#r28811098>.
>